### ISSUE TITLE
Add new option in rows_by_key to return scalar values when unique=True

### ISSUE THREAD

**Issue by CaselIT** [pola-rs/polars#19994]
### Description

Before polars 1.14 when running `rows_by_key(columns, unique=True)` if the resulting resulting row included only a single column the returned value was a scalar.
In 1.15 it's always a tuple. Since the new behaviour is what was always documented I consider this a bugfix.

In any case this feature was useful, so I'm proposing a new flag, like `rows_by_key(..., scalars=True)` or a new value for unique, like `rows_by_key(..., unique='scalars')` that restores the 1.14 behaviour. This flag could either validate that the return are indeed 1-tuple or always return the first index of the tuple.

An example follow:
This simple script
```py
import polars as pl

df = pl.DataFrame({"a": [1, 2, 3], "b": [4, 5, 5], "c": [7, 8, 9]})

print("2 columns", df.rows_by_key("a", unique=True))
print("1 column", df.select("a", "c").rows_by_key("a", unique=True))
```
prints in 1.14 the following:
```
2 columns {1: (4, 7), 2: (5, 8), 3: (5, 9)}
1 column {1: 7, 2: 8, 3: 9}
```
while in 1.15 it prints:
```
2 columns {1: (4, 7), 2: (5, 8), 3: (5, 9)}
1 column {1: (7,), 2: (8,), 3: (9,)}
```

**Comment by CaselIT** [comment]
Updated description and title since this affect also when unique is not selected

**Comment by CaselIT** [comment]
As a note, this change does not seem listed in the changelog for 1.15

**Comment by Julian-J-S** [comment]
yeeeees! 
Had the same problem. Broke a test pipeline 😞

**Comment by CaselIT** [comment]
same here

**Comment by gab23r** [comment]
I agree that this would be nice, however I would see another api, we can add a `subset : list[str] | str | None` argument.
if subset is a list or is None we return tuples, if it is a single column we return scalar :

```python
df = pl.DataFrame({"a": [1, 2, 3], "b": [4, 5, 5], "c": [7, 8, 9]})

df.rows_by_key("a", unique=True))
# {1: (4, 7), 2: (5, 8), 3: (5, 9)}

df.rows_by_key("a", unique=True, subset="c"))
# {1: 7, 2: 8, 3: 9}

df.rows_by_key("a", unique=True, subset=["c"]))
# {1: (7,), 2: (8,), 3: (9,)}



```

**Comment by gab23r** [comment]
I propose to implement if accepted

**Comment by CaselIT** [comment]
@gab23r selecting the column is already available via `select()` so there may be redundancy there.
But either would work for me as api.

**Comment by lorentzenchr** [comment]
This also effected my pipeline. Does someone know, which PR changed the behaviour?

Could the docs of [`rows_by_key`](https://docs.pola.rs/api/python/stable/reference/dataframe/api/polars.DataFrame.rows_by_key.html#polars.DataFrame.rows_by_key) be improved by a clear statement of the type of values one gets, i.e. values are lists of tuples?
Does someone know the reason why this tuple is needed. To me, it seems redundant (or the list is).

**Comment by cmdlineluser** [comment]
@lorentzenchr It seems to have changed with https://github.com/pola-rs/polars/pull/19486

**Comment by SamuelAllain** [comment]
+1 this change also broke our pipeline. I vote for the "scalars" option which I find clearer than the "subset" one
